### ISSUE TITLE
FormFactory

### ISSUE THREAD

**Issue by harikt** [auraphp/Aura.Input#22]
Can we add a FormFactory to the `default.php`

``` php
<?php
$di->params['Aura\Input\FormFactory'] = [
    // a map of form names to form factories
    'map' => [
        'map-name' => $di->newFactory('classes'),
    ],
];

$di->set('form_factory', $di->lazyNew('Aura\Input\FormFactory'));

$di->setter['Aura\Framework\Web\Controller\AbstractPage']['setFormFactory'] = $di->lazyGet('form_factory');
```

and the `FormFactory` as 

``` php
<?php
namespace Aura\Input;

class FormFactory
{
    protected $map = [];

    public function __construct($map = [])
    {
        $this->map = $map;
    }

    public function newInstance($name)
    {
        if (! array_key_exists($name, $this->map)) {
            throw new \Exception("No form with the name found");
        }
        $factory = $this->map[$name];
        $form = $factory();
        return $form;
    }
}
```

That way everyone don't need to spend more time on making things to work. What do you think ? I can give a PR if so.

**Comment by pmjones** [comment]
Yeah, if you want to send a PR for this, go for it.

**Comment by harikt** [comment]
@pmjones this was an idea some months back, but later I thought it is good to have a generic factory . See what I added at http://harikt.com/the-book-on-aura/chapter-9.html and http://harikt.com/the-book-on-aura/chapter-13.html . Seems a generic factory will help just not forms.

**Comment by pmjones** [comment]
Hm. "Generic Factory" looks an awful lot like a service locator, at least the object-creation part of a service locator. That's the kind of thing we should try to stay away from. If we must have service locators, they should be limited to one type of object, not many different types of objects (e.g., just forms, not forms + models + connections + whatever else).

**Comment by pmjones** [comment]
Fixed as of f78d8ad95611e13e65b8ce2be546951869cec3f0

**Comment by harikt** [comment]
Thank you @pmjones  :)
